content/code/testing.md

```diff
@@ -104,6 +150,20 @@ This can be very useful, as it enables creating a hierarchical structure within
 {!index_internal_test.md!}
     ```
 
+##### Structure
+
+In those tests, usually the structure is composed of:
+
+* At the top-level, the common testing variables that are used in multiple subtests.
+    * This can be a logger that discards logs, some test input that can be reused, expected return values shared between tests, and so on.
+* In each sub-test:
+    * The test is defined as parallelized;
+    * The inputs and outputs are defined in variables;
+    * The mocks are defined and their methods overridden when necessary;
+    * The test subject is created and given the mocks;
+    * The tested method is called on the subject, its return values are often stored in `got` and `err`: `got, err := mything.Do(input)`;
```

Comment:
What is received from the test call is more like an expected value, that we also want to receive from the function call that is tested and compared against.
I would replace the word `got` with `expected`

Reply:
No, this is about the actual call which is being tested.

What you are talking about refers to what you do in your test:

```go
expected := myMock.Retrieve()

got := subject.Retrieve()

assert.EqualElements(t, expected, got)
```

Which is an edge case that only happens because you test that the subject returns whatever the mock returns without any modification.

In my opinion it's not the best practice either, I would personally rather have a hardcoded `want` value and make the mock return a hardcoded value as well, because this test relies on the mock to get its output variable which is very uncommon and reduces clarity IMO.

I'm fine with it in the context of your test as long as it remains localized and does not become a habit/shortcut, but it should definitely not be in the docs IMO.

WDYT?

Reply:
Yes, I saw that this is about the tested function call.